src/core/ext/xds/xds_client.cc

```diff
@@ -1066,15 +1066,17 @@ void XdsClient::ChannelState::AdsCallState::OnRecvMessage(
                 }
               } else {
                 resource_state.resource.reset();
+                resource_state.meta.client_status =
+                    XdsApi::ResourceMetadata::DOES_NOT_EXIST;
                 xds_client()->NotifyWatchersOnResourceDoesNotExist(
                     resource_state.watchers);
               }
             }
           }
         }
       }
-      // If we had valid resources, update the version.
-      if (result.have_valid_resources) {
+      // If we had valid resources or the update was empty, update the version.
+      if (result.have_valid_resources || result.errors.empty()) {
```

Comment:
is an empty update allowed?

Reply:
Yes.  For the LDS and CDS resources in SotW, this will happen if every resource that the client is subscribed to is deleted on the server.